utils/oscap-info.c

```diff
@@ -89,21 +91,103 @@ static inline void _print_xccdf_status(struct xccdf_status *status, const char *
+		xccdf_profile_get_id(prof),
+		xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+}
+
+static void _print_xccdf_profile_verbose(const struct xccdf_profile *prof, const char *prefix)
+{
+	struct oscap_text_iterator *text_it;
+	struct oscap_string_iterator *str_it;
+
+	printf("%sProfile\n", prefix);
+
+	text_it = xccdf_profile_get_title(prof);
+	char *profile_title = oscap_textlist_get_preferred_plaintext(text_it, NULL);
+	oscap_text_iterator_free(text_it);
+	_remove_occurence_of_character_from_string(profile_title, '\n');
+	printf("%s\tTitle: %s\n", prefix, profile_title);
+	free(profile_title);
+
+	printf("%s\tId: %s%s\n", prefix,
+		xccdf_profile_get_id(prof),
+		xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+
+	text_it = xccdf_profile_get_description(prof);
+	char *profile_description = oscap_textlist_get_preferred_plaintext(text_it, NULL);
+	oscap_text_iterator_free(text_it);
+	_remove_occurence_of_character_from_string(profile_description, '\n');
+	printf("%s\tDescription: %s\n", prefix, profile_description);
+	free(profile_description);
+
+	str_it = xccdf_profile_get_platforms(prof);
+	printf("%s\tPlatforms: ", prefix);
+	while (oscap_string_iterator_has_more(str_it)) {
+		const char *platform = oscap_string_iterator_next(str_it);
+		printf("%s,", platform);
+	}
+	printf("\n");
+	oscap_string_iterator_free(str_it);
+}
+
+static void _print_xccdf_profile_terse(const struct xccdf_profile *prof, const char *prefix)
+{
+	struct oscap_text_iterator *title_it = xccdf_profile_get_title(prof);
+	char *profile_title = oscap_textlist_get_preferred_plaintext(title_it, NULL);
+	_remove_occurence_of_character_from_string(profile_title, '\n');
+	oscap_text_iterator_free(title_it);
+	const char *profile_id = xccdf_profile_get_id(prof);
+	char cleaned_profile_id[1024];
+	strncpy(cleaned_profile_id, profile_id, sizeof(cleaned_profile_id) - 1);
+	_remove_occurence_of_character_from_string(cleaned_profile_id, '\n');
+	printf("%s:%s\n",
+		cleaned_profile_id,
+		profile_title);
+	free(profile_title);
+}
+
+static void _print_xccdf_profile_with_id(struct xccdf_profile_iterator *prof_it, const char *profile_id, const char *prefix, void (*print_one_profile)(const struct xccdf_profile *, const char *))
+{
+	if (print_one_profile == NULL)
+		print_one_profile = &_print_xccdf_profile_default;
 	while (xccdf_profile_iterator_has_more(prof_it)) {
-		struct xccdf_profile * prof = xccdf_profile_iterator_next(prof_it);
-		struct oscap_text_iterator *title_it = xccdf_profile_get_title(prof);
-		char *profile_title = oscap_textlist_get_preferred_plaintext(title_it, NULL);
-		oscap_text_iterator_free(title_it);
-		printf("%s\tTitle: %s\n", prefix, profile_title);
-		free(profile_title);
-		printf("%s\t\tId: %s%s\n", prefix,
-			xccdf_profile_get_id(prof),
-			xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+		struct xccdf_profile *prof = xccdf_profile_iterator_next(prof_it);
+		if (strncmp(xccdf_profile_get_id(prof), profile_id, 1024) == 0)
```

Comment:
Why is this 1024? Shouldn't it be `strlen(profile_id)` instead?

Reply:
I don't know whether to trust the arguments to be 0-terminated. The code above is safe and will work correctly all right if profile IDs have less than 1024 characters. I am happy to learn a better way how to handle the situation; however if those strings are safe, than I would use `strcmp` without worrying about length.

Reply:
They're safe, you don't have to handle this